<launch>
  <node name="Operator" pkg="nav2d_operator" type="operator" >
    <remap from="scan" to="/ropod/laser/scan"/>
    <rosparam file="$(find nav2d_operator)/config/operator.yaml"/>
    <rosparam file="$(find nav2d_operator)/config/costmap.yaml" ns="local_map" />
  </node>

  <node name="corridor_navigation" pkg="corridor_navigation" type="corridor_navigation_node">
    <param name="gateway_detection_topic" value="/gateways_detected" />
    <param name="distance_monitor_topic" value="/monitored_distance" />
    <param name="heading_monitor_topic" value="/monitored_heading" />
    <param name="desired_heading_topic" value="/desired_heading" />
    <param name="desired_velocity_topic" value="/desired_velocity" />
    <param name="nav2d_operator_cmd_topic" value="/cmd" />

    <param name="reset_distance_monitor_service" value="/reset_distance_monitor" />
    <param name="reset_heading_monitor_service" value="/reset_heading_monitor" />
    <param name="heading_control_switch_service" value="/heading_control_switch" />

    <param name="recovery_direction_threshold" value="0.5" />
    <param name="correction_direction_threshold" value="0.06" />

    <param name="controller_frequency" value="20" />
    <param name="velocity" value="0.5" />
  </node>
</launch>

